packages/js/src/plugins/auctionHouseModule/updateAuctionHouse.ts

```diff
@@ -149,4 +162,55 @@ export const updateAuctionHouseBuilder = (
         key: params.instructionKey ?? 'updateAuctionHouse',
       })
     );
+
+  if (params.auctioneerAuthority) {
+    const auctioneerAuthority = params.auctioneerAuthority as PublicKey;
+    const ahAuctioneerPda = findAuctioneerPda(
+      auctionHouse.address,
+      auctioneerAuthority
+    );
+
+    const scopes = params.auctioneerScopes ?? [
+      AuthorityScope.Deposit,
+      AuthorityScope.Buy,
+      AuthorityScope.PublicBuy,
+      AuthorityScope.ExecuteSale,
+      AuthorityScope.Sell,
+      AuthorityScope.Cancel,
+      AuthorityScope.Withdraw,
+    ];
+    const instructionAccounts = {
+      auctionHouse: auctionHouse.address,
+      authority: authority.publicKey,
+      auctioneerAuthority,
+      ahAuctioneerPda,
+    };
+
+    const ahAuctioneerAccount = await metaplex
+      .rpc()
+      .getAccount(ahAuctioneerPda);
+
+    // Run Auctioneer update if it has been delegated.
+    if (auctionHouse.hasAuctioneer && ahAuctioneerAccount.exists) {
```

Comment:
I still think this is unnecessary. We're making an extra HTTP call to check the content of any account when the `AuctionHouse` account already tells us if the Auction House is delegating to Auctioneer or not via `auctionHouse.hasAuctioneer`. Can you explain to me why we need to make an extra HTTP call to check the account's content? In other words, how is it possible to have `auctionHouse.hasAuctioneer` set to `true`, yet, without delegating to an Auctioneer instance?

I'm suggesting the following which means we can removing remove the async/await of the builder method.

```suggestion
    // Run Auctioneer update if it has been delegated.
    if (auctionHouse.hasAuctioneer) {
```

Reply:
It covers the case when we delegate new auctioneer authority. Since the account for it isn't created, updateAuctioneerInstruction won't work, because it utilizes the Auctioneer account in the program.

Reply:
But when we update the Auction House model, we return a fresh new instance of that model specifically for that.

We can't be doing extra HTTP calls just in case the provided model is not on par with what's on-chain because otherwise we might as well not require the `AuctionHouse` model at all and fetch it ourselves.

Reply:
But how can I check if is this a new auctioneer authority to delegate or auctioneer authority that has been already delegated to AH?

I have a test that covers this scenario.

1. Create an AH.
2. Delegate auctioneerAuthority №1
3. Call updateAuctionHouse with auctioneerAuthority №2. 
4. Since there is no account for auctioneerAuthority №2 PDA -> Delegate this new auctioneerAuthority

Reply:
Oh, my bad I understand now. Usually, I would have said, we need an additional `newAuctioneerAuthority` parameter (like we do with other `update` operations on the SDK like when we update an NFT) but in this particular case, you're right, it makes more sense to fetch the content of the account.

And the reason for that is related to my comment below. We can fetch the content of the Auctioneer PDA to see the scopes and make sure they are not overridden.

Reply:
Actually no wait, we need both. We need the current `auctioneerAuthority` to fetch the current scopes and `newActioneerAuthority` to update the auctioneer instance using the same scopes.